empire.go

```diff
@@ -141,6 +145,10 @@ func (opts CreateOpts) Event() CreateEvent {
 
 // Create creates a new app.
 func (e *Empire) Create(ctx context.Context, opts CreateOpts) (*App, error) {
+	if err := e.MaintenanceMode; err != nil {
```

Comment:
We can probably start with that. I wanted to put this in the core API that the control layers consume so that it wouldn't get duplicated. If we put it in the control layers, than we have to duplicate it for the heroku api, github webhooks and slack commands (if we ever merge that PR in).

I think I'm still mostly in favor of just doing what I'm doing right now. You have to think about adding the maintenance mode check if you add a new method to the core API, but honestly, that interface is pretty stable and doesn't change very frequently.